src/VerifyPN.cpp

```diff
@@ -671,3 +671,138 @@ void simplify_queries(  const MarkVal* marking,
             return a;
     }) && std::chrono::duration_cast<std::chrono::seconds>(end - begin).count() < options.queryReductionTimeout && to_handle > 0);
 }
+
+void simplify_queries_potency(const MarkVal* marking,
```

Comment:
Is this code duplicated from somewhere ? It looks like it.

Reply:
Yes, it is the function simplify_queries with less code, and with the potencies = [...] line added

Reply:
I could add 2 optional parameters to the simplify_queries function (bool isLp and initPotencies), and then do the correct things according to the bool isLP